server/server.go

```diff
@@ -1411,11 +1411,14 @@ func (s *StanServer) buildServerURLs() ([]string, error) {
 		// use 127.0.0.1 or ::1 for host address (important for
 		// Windows since connect with 0.0.0.0 or :: fails).
 		sport := strconv.Itoa(opts.Port)
-		if opts.Host == "0.0.0.0" {
+		switch opts.Host {
+		case "0.0.0.0":
 			hostport = net.JoinHostPort("127.0.0.1", sport)
-		} else if opts.Host == "::" || opts.Host == "[::]" {
+		case "::":
 			hostport = net.JoinHostPort("::1", sport)
-		} else {
+		case "[::]":
```

Comment:
I think what @ripienaar was saying is that you could have the `"::"` and `[::]` in the same case statement:
```
switch opts.Host {
    case "0.0.0.0":
	hostport = net.JoinHostPort("127.0.0.1", sport)
    case "::", "[::]":
	hostport = net.JoinHostPort("::1", sport)
    default:
	hostport = net.JoinHostPort(opts.Host, sport)
}
```

Reply:
True, I changed it.